content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,563 @@
+
+```python
+# load the data
+
+train_data = pickle.load(open("./data/train.p", "rb"))  # 728,789 examples
+valid_data = pickle.load(open("./data/valid.p", "rb"))  # 182,198 examples
+test_data = pickle.load(open("./data/test.p", "rb"))  # 100,000 examples
+```
+
+As we explained before, the (train) dataset is a list of `728,789` tuples where the first element is the full address, and the second is a list of the tag (the ground truth).
+
+```python
+train_data[0:2]
+```
+
+> (the output)
+
+![data_snapshot](data_snapshot.png)
+
+### Vectorize the Dataset
+
+Since we used word embeddings as our encoded representation of the words, we need to *convert* the address into word vector. In order to do that, we will use a vectorizer. This embedding vectorizer will extract, for each word, the embedding value based on the pre-trained French fastText model.
+
+```python
+class EmbeddingVectorizer:
+    def __init__(self):
+        """
+        Embedding vectorizer
+        """
+        fasttext.util.download_model('fr', if_exists='ignore')
+        self.embedding_model = fasttext.load_model("./cc.fr.300.bin")
+
+    def __call__(self, address):
+        """
+        Convert address to embedding vectors
+        :param address: The address to convert
+        :return: The embeddings vectors
+        """
+        embeddings = []
+        for word in address.split():
+            embeddings.append(self.embedding_model[word])
+        return embeddings
+     
+embedding_vectorizer = EmbeddingVectorizer()
+```
+
+We also need to apply a similar operation to the address tags (e.g. StreeNumber, StreetName). 
+This time, however, the vectorizer needs to convert the tags into categorical values (e.g. 0, 1). 
+
+For simplicity, we will use a `DatasetVectorizer` class that will apply the vectorizing process using both 
+the embedding and the address vectorize process.
+
+```python
+class DatasetVectorizer:
+    def __init__(self, embedding_vectorizer):
+        self.embedding_vectorizer = embedding_vectorizer
+        self.tags_set = {
+            "StreetNumber": 0,
+            "StreetName": 1,
+            "Unit": 2,
+            "Municipality": 3,
+            "Province": 4,
+            "PostalCode": 5,
+            "Orientation": 6,
+            "GeneralDelivery": 7
+        }
+
+    def vectorize(self, data):  # We vectorize inplace
+        for idx, item in enumerate(data):
+            data[idx] = self._item_vectorizing(item)
+
+    def _item_vectorizing(self, item):
+        address = item[0]
+        address_vector = self.embedding_vectorizer(address)
+
+        tags = item[1]
+        idx_tags = self._convert_tags_to_idx(tags)
+
+        return address_vector, idx_tags
+
+    def _convert_tags_to_idx(self, tags):
+        idx_tags = []
+        for tag in tags:
+            idx_tags.append(self.tags_set[tag])
+        return idx_tags
+
+dataset_vectorizer = DatasetVectorizer(embedding_vectorizer)
+```  
+
+Now, let's vectorize (in place) our dataset.
```

Comment:
Je suis trop newb pour savoir ce que "in place" signifie.

Reply:
Je l'ai retirer n'apporte rien.